### ISSUE TITLE
produced POM for gradle plugin portal doesn't use resolved dependencies

### ISSUE THREAD

**Issue by rpalcolea** [nebula-plugins/gradle-nebula-integration#60]
Hi folks, we detected an issue with our plugins getting published to plugin portal without the resolved dependency version for each one of them. We were expecting to configure the `MavenPublication` once with our plugin but things didn't work. This is here -> https://github.com/nebula-plugins/nebula-publishing-plugin/blob/master/src/main/groovy/nebula/plugin/publishing/maven/MavenResolvedDependenciesPlugin.groovy

With that said, we tried to reproduce this without Nebula and just vanilla Gradle as follows: https://github.com/nebula-plugins/gradle-nebula-integration/blob/master/gradle-plugin-pom-resolved-dependencies/build.gradle 

We configure the publication to use resolved dependencies: 

```
publishing {
    publications {
        maven(MavenPublication) {
            from components.java
            group 'nebula.test.plugin'
            version '1.0.0'
            versionMapping {
                allVariants {
                    fromResolutionResult()
                }
            }
        }
    }
}
```

When we publish this plugin to maven local, we get the following POM:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>nebula.test.plugin</groupId>
  <artifactId>gradle-plugin-test</artifactId>
  <version>1.0.0</version>
  <dependencies>
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>27.1-jre</version>
      <scope>compile</scope>
    </dependency>
  </dependencies>
</project>
```

However, when trying to push this repo to the plugin portal, this is the output:

```
> Task :publishPlugins
Publishing plugin nebula.test-nothing-plugin version 1.0.0
Thank you. Your new plugin nebula.test-nothing-plugin has been submitted for approval by Gradle engineers. The request should be processed within the next 24 hours, at which point you will be contacted via email.
Publishing artifact build/libs/gradle-plugin-test-1.0.0.jar
Publishing artifact build/libs/gradle-plugin-test-1.0.0-sources.jar
Publishing artifact build/libs/gradle-plugin-test-1.0.0-javadoc.jar
Publishing artifact build/libs/gradle-plugin-test-1.0.0-groovydoc.jar
Publishing artifact build/publish-generated-resources/pom.xml
Activating plugin nebula.test-nothing-plugin version 1.0.0
```

And the generated `pom.xml` is:

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0">
  <modelVersion>4.0.0</modelVersion>
  <groupId>gradle.plugin.nebula.test.plugin</groupId>
  <artifactId>gradle-plugin-test</artifactId>
  <version>1.0.0</version>
  <dependencies>
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>latest.release</version>
      <scope>compile</scope>
      <type>jar</type>
      <optional>false</optional>
    </dependency>
  </dependencies>
</project>
```

It appears that any change to the `MavenPublication` is not honored by the plugin-publish-plugin

**Comment by ljacomet** [comment]
I believe this was resolved in gradle/gradle#8879

**Comment by rpalcolea** [comment]
Correct, closing now
